### PR TITLE
Allow ud_grade to work with read only maps.

### PR DESCRIPTION
**Pull request by cjcopi**
Work around for issue #262.

### CODE REVIEW

File: healpy/pixelfunc.py [hunk 1]
```diff
@@ -1647,10 +1647,9 @@ def _ud_grade_core(m,nside_out,pess=False,power=None, dtype=None):
         bads = (mask_bad(m) | (~np.isfinite(m)))
         hit = np.ones(npix_in,dtype=np.int16)
         hit[bads] = 0
```
Comment (lpsinger) [inline healpy/pixelfunc.py]: Note that `hit` is equivalent to `~bads`.

**Comment by cjcopi** [comment]
Updated to remove hit.  I changed bads to goods since that is what is actually wanted and it can be calculated from the reshaped array (mr).  Since rat2 is an integer I also enforce integer division.  The logic of the goods assignment could be "multiplied out" but I have not done so.

**Comment by lpsinger** [comment]
:+1:

**Comment by zonca** [comment]
great thanks!

**Pull request merged**